app/src/main/java/com/parishod/watomatic/model/utils/AutoStartHelper.java

```diff
@@ -0,0 +1,401 @@
+            case BRAND_ASUS:
+                autoStartAsus(context);
+                break;
+            case BRAND_XIAOMI:
+            case BRAND_XIAOMI_POCO:
+            case BRAND_XIAOMI_REDMI:
+                autoStartXiaomi(context);
+                break;
+            case BRAND_LETV:
+                autoStartLetv(context);
+                break;
+            case BRAND_HONOR:
+                autoStartHonor(context);
+                break;
+            case BRAND_HUAWEI:
+                autoStartHuawei(context);
+                break;
+            case BRAND_OPPO:
+                autoStartOppo(context);
+                break;
+            case BRAND_VIVO:
+                autoStartVivo(context);
+                break;
+            case BRAND_NOKIA:
+                autoStartNokia(context);
+                break;
+            case BRAND_SAMSUNG:
+                autoStartSamsung(context);
+                break;
+            case BRAND_ONE_PLUS:
+                autoStartOnePlus(context);
+                break;
+        }
+
+    }
+
+    private void autoStartSamsung(final Context context) {
+        String packageName;
+        if (Build.VERSION.SDK_INT > Build.VERSION_CODES.N)
+            packageName = PACKAGE_SAMSUNG_MAIN1;
+        else
+            packageName = PACKAGE_SAMSUNG_MAIN2;
+
+        if (isPackageExists(context, packageName)) {
+
+            showAlert(context, (dialog, which) -> {
+                try {
+                    startIntent(context, packageName, PACKAGE_SAMSUNG_COMPONENT1);
+                } catch (Exception e) {
+                    e.printStackTrace();
+                    try {
+                        startIntent(context, packageName, PACKAGE_SAMSUNG_COMPONENT2);
+                    } catch (Exception e1) {
+                        e1.printStackTrace();
+                    }
+                }
+                dialog.dismiss();
+            });
+
+        }
+    }
+
+    private void autoStartAsus(final Context context) {
+        if (isPackageExists(context, PACKAGE_ASUS_MAIN)) {
+
+            showAlert(context, (dialog, which) -> {
+                try {
+                    startIntent(context, PACKAGE_ASUS_MAIN, PACKAGE_ASUS_COMPONENT);
+                } catch (Exception e) {
+                    e.printStackTrace();
+                    try {
+                        startIntent(context, PACKAGE_ASUS_MAIN, PACKAGE_ASUS_COMPONENT_FALLBACK);
+                    } catch (Exception e1) {
+                        e1.printStackTrace();
+                    }
+                }
+                dialog.dismiss();
+            });
+
+        }
+    }
+
+    private void showAlert(Context context, DialogInterface.OnClickListener onClickListener) {
+        CustomDialog customDialog = new CustomDialog(context);
+        Bundle bundle = new Bundle();
+        bundle.putString(Constants.PERMISSION_DIALOG_TITLE, context.getString(R.string.auto_start_permission_dialog_title));
```

Comment:
`R.string.auto_start_permission_dialog_title` and other string are used here.

These are also not needed right? Can we also remove `showAlert()` then?
Can you clean this up if possible @spuday90

Reply:
hmm the popup does not show up on emulators. Clicking on the preference in "Settings" does not do anything on emulators. But it shows this popup on my real device.

Reply:
We are showing the popup in settings page